h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoReader.java

```diff
@@ -0,0 +1,102 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.ModelMojoReader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TargetEncoderMojoReader extends ModelMojoReader<TargetEncoderMojoModel> {
+
+  @Override
+  public String getModelName() {
+    return "TargetEncoder";
+  }
+
+  @Override
+  protected void readModelData() throws IOException {
+    _model._withBlending = readkv("with_blending");
+    if(_model._withBlending) {
+      _model._inflectionPoint = readkv("inflection_point");
+      _model._smoothing = readkv("smoothing");
+    }
+    _model._targetEncodingMap = parseEncodingMap("feature_engineering/target_encoding/encoding_map.ini");
+  }
+
+  @Override
+  protected TargetEncoderMojoModel makeModel(String[] columns, String[][] domains, String responseColumn) {
+    return new TargetEncoderMojoModel(columns, domains, responseColumn);
+  }
+  
+  public EncodingMaps parseEncodingMap(String pathToSource) throws IOException {
+    Map<String, EncodingMap> encodingMaps = null;
+
+    if(getMojoReaderBackend().exists(pathToSource)) {
+      BufferedReader source = getMojoReaderBackend().getTextFile(pathToSource);
```

Comment:
I think it is better to use the exposed API: exists and readtext, readtext also handles unescaping of newlines that might be found in categorical level names

Reply:
@michalkurka Thanks, good to know that we have delegating methods in ModelMojoReader.